ql/ql/src/queries/style/FieldOnlyUsedInCharPred.ql

```diff
@@ -0,0 +1,25 @@
+/**
+ * @name Field only used in CharPred
+ * @description A field only being used in the charpred suggests that the field should be inlined into the charpred.
+ * @kind problem
+ * @problem.severity recommendation
+ * @id ql/field-only-used-in-charpred
+ * @tags maintainability
+ * @precision very-high
+ */
+
+import ql
+
+from Class c, FieldDecl f
+where
+  c.getAField() = f and
+  forex(FieldAccess access | access.getDeclaration() = f |
+    access.getEnclosingPredicate() = c.getCharPred()
+  ) and
+  // excluding fields that are uniquely used in call to an IPA constructor
+  not unique(FieldAccess access | access.getDeclaration() = f | access) =
```

Comment:
Could you explain this (line 20-23) to me, perhaps in the huddle for a faster turnaround time? It reads a bit like you might exclude cases where the field is used multiple times in the charpred. But maybe that's not an issue here, because we're looking at IPA constructors only? Anyway, it's a bit opaque to me.

Reply:
Yes, I'm excluding cases where the field is used multiple times in the charpred.  

It's to only exclude the pattern where the field is only used as an argument to a call to an IPA constructor. 
That pattern is quite common, and perfectly readable, so we decided to exclude it.  

You can see the pattern a lot in [`Ast.qll` in QL-for-QL](https://github.com/github/codeql/blob/6aad8d6897e8b24790a8be1338656c81e80f10f5/ql/ql/src/codeql_ql/ast/Ast.qll#L511).

Reply:
OK, we also had a chat about this. Makes sense to me. 👍